DEPLOY.rst

```diff
@@ -0,0 +1,198 @@
+        opendebates:
+          production: opendebates-production-ag
+          staging: opendebates-staging-ag
+          testing: opendebates-testing-ag
+
+and change the appropriate value to the name of the autoscaling group you
+created.  (Or create it with that name to begin with.)
+
+Automation commands
+~~~~~~~~~~~~~~~~~~~
+
+Now you can run this command to set up everything else::
+
+    fab create_environment:opendebates,<environment>
+
+It'll take a while (30 minutes? an hour?) but you only need to do it
+once per environment.
+
+Updating code
+-------------
+
+This is what to do when the code changes and you want the servers in
+an environment to switch to the newer code.
+
+Read through this whole section before starting to update anything,
+please!
+
+Step 1: Create a new launch configuration. This is a saved EC2 instance image
+that the autoscaler uses to spin up new web servers::
+
+     fab create_launch_config_for_deployment:opendebates,<environment>
+
+On an m1.small instance, this'll take just over 20 minutes. It might be faster
+if your web servers are using a faster server.
+
+At the end of the output of that command, it'll print out a long string that
+is the name of the new launch configuration. Save that somewhere.
+
+Now, which order you do steps 2 and 3 in might depend on whether you
+have migrations and whether it's okay for the migration to run before
+the web servers have been updated, or for the web servers to be updated
+before the migrations are run, or whether they really really need to happen
+at the same time.
+
+If you can run migrations first, or don't have any, do step 2, then step 3.
+
+If you have to update webservers first, do step 3, then step 2.
+
+If you really have to update things all together, skip down a bit
+to "Updating everything at once".
+
+Step 2: Update your worker(s)::
+
+    fab <environment> deploy_worker
+
+This will also run migrations.
+
+Step 3: If the changes are minor, such that it's okay if some servers are
+temporarily running the newer code while others are still running the older,
+then you can update the web servers without downtime::
+
+    fab deploy_serial:opendebates,<environment>,<launch config name>
+
+This will take the web servers down one at a time and bring up a replacement,
+waiting each time until the replacement is healthy before doing the next. It'll
+take quite a while if there are many servers, but no downtime is needed.
+
+If you need to update things all at the same time, some downtime will be needed.
+On the bright side, this command will run much faster than the other one.
+Here it is::
+
+    fab deploy_full:opendebates,<environment>,<launch config name>
+
+This'll put up an "upgrade in progress" notice on the site, take down all the
+webservers, create new ones using the new launch config, and take down the
+upgrade notice again once everything looks okay. In my testing it took about
+4 minutes each time.
+
+Updating everything at once
```

Comment:
This is what deploy_full does; this section should be unnecessary. Note that deploy_full also has some logic to ensure that, in case the repo has been updated since you created your launch config, it still deploys the correct version (i.e., the version in use in the launch config) of the code.